blackbird_python/blackbird/program.py

```diff
@@ -314,25 +313,46 @@ def serialize(self):
 
         script = ["name {}".format(self.name), "version {}".format(self.version)]
 
-        if self.target["name"] is not None:
-            array_insert += 1
-            options = ""
+        # add target and type to the script
+        for name, data in [("target", self.target), ("type", self.programtype)]:
+            if data["name"] is not None:
+                array_insert += 1
+                options = ""
 
-            if self.target["options"]:
-                # if the target has options, compile them into
-                # the expected syntax
-                option_strings = [
-                    "{}={}".format(k, v) if not isinstance(v, str) else '{}="{}"'.format(k, v)
-                    for k, v in self.target["options"].items()
-                ]
-                options = " ({})".format(", ".join(option_strings))
+                if data["options"]:
+                    # if the target has options, compile them into
+                    # the expected syntax
+                    option_strings = [
+                        "{}={}".format(k, v) if not isinstance(v, str) else '{}="{}"'.format(k, v)
+                        for k, v in data["options"].items()
+                    ]
```

Comment:
This list comprehension seems a bit long, maybe good to go with a `for` loop here?

Reply:
Oh, I think I just like list comprehensions a bit too much, but sure, a `for` loop could also be used. :smile: